src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseFVM.cpp

```diff
@@ -103,6 +103,25 @@ void SinglePhaseFVM< BASE >::setupSystem( DomainPartition & domain,
                      solution,
                      setSparsity );
 
+  if( !m_precond && m_linearSolverParameters.get().solverType != LinearSolverParameters::SolverType::direct )
+  {
+    m_precond = createPreconditioner( domain );
+  }
+}
+
+template< typename BASE >
+std::unique_ptr< PreconditionerBase< LAInterface > >
+SinglePhaseFVM< BASE >::createPreconditioner( DomainPartition & domain ) const
+{
+  LinearSolverParameters const & linParams = m_linearSolverParameters.get();
+  GEOS_UNUSED_VAR( domain );
+  switch( linParams.preconditionerType )
+  {
+    default:
+    {
+      return LAInterface::createPreconditioner( linParams );
+    }
```

Comment:
The `switch` statement may seem pointless at the moment, but it's left as a placeholder for adding custom preconditioners (e.g. multiscale)